samza-core/src/test/java/org/apache/samza/task/TestStreamOperatorTask.java

```diff
@@ -80,4 +84,34 @@ public void testExceptionsInProcessInvokesTaskCallback() throws InterruptedExcep
         mockTaskCoordinator, mockTaskCallback);
     failureLatch.await();
   }
+
+  /**
+   * Tests if the appropriate SamzaException is propagated to the TaskCallback if there is no InputOperator for a given
+   * SystemStream in the OperatorGraph.
+   * */
+  @Test
+  public void testExceptionIfInputOperatorMissing() throws NoSuchFieldException, IllegalAccessException {
+    IncomingMessageEnvelope mockIme = mock(IncomingMessageEnvelope.class, RETURNS_DEEP_STUBS);
+    SystemStream testSystemStream = new SystemStream("foo", "bar");
+    when(mockIme.getSystemStreamPartition().getSystemStream()).thenReturn(testSystemStream);
+
+    OperatorImplGraph mockOperatorImplGraph = mock(OperatorImplGraph.class);
+    when(mockOperatorImplGraph.getInputOperator(anyObject())).thenReturn(null);
+    // OperatorImplGraph is a private field ('operatorImplGraph') in StreamOperatorTask and need to be set with
+    // the mocked value reflectively
+    StreamOperatorTask operatorTask = new StreamOperatorTask(mock(OperatorSpecGraph.class));
+    Field operatorImplGraphField = operatorTask.getClass().getDeclaredField("operatorImplGraph");
+    operatorImplGraphField.setAccessible(true);
+    operatorImplGraphField.set(operatorTask, mockOperatorImplGraph);
+
+    TaskCallback mockTaskCallback = mock(TaskCallback.class);
+    operatorTask.processAsync(mockIme, mock(MessageCollector.class), mock(TaskCoordinator.class), mockTaskCallback);
+    // ArgumentCaptor for capturing the SamzaException passed to the callback
+    ArgumentCaptor<Throwable> throwableCaptor = ArgumentCaptor.forClass(Throwable.class);
+    verify(mockTaskCallback, only()).failure(throwableCaptor.capture());
+    assertEquals(throwableCaptor.getValue().getClass(), SamzaException.class);
+    assertEquals(
+        throwableCaptor.getValue().getMessage(),
+        String.format("InputOperator not found in OperatorGraph for %s", testSystemStream.toString()));
```

Comment:
same as above. `testSystemStream` should be sufficient for the `String.format(...)` call.